apis/management.cattle.io/v3/authn_types.go

```diff
@@ -80,6 +80,7 @@ type UserAttribute struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	UserName        string
+	UserInfo        map[string]string
```

Comment:
We could store the the userPrincipal v3.Principal directly here instead of another map[string]string. Also if there is a need to store attributes for more than one auth provider in future, storing the principal per provider will be helpful.